client/client.go

```diff
@@ -5,7 +5,7 @@ import (
 	"errors"
 	"time"
 
-	"github.com/drand/drand/key"
+	"github.com/drand/drand/chain"
```

Comment:
This is pulling in a lot of stuff that ideally we wouldn't need in the client library.
chain depends on `net`, `grpc`, `bolt`. Is there a way to not pull those in?

Reply:
I wanted to avoid doing all the splitting within the same PR but that's the goal: `chain/` would only contains the higher level structs and util methods (chain info, beacon, time) and `chain/store` would contain the store definition and `chain/protocol` would contain the protocol. (or a better name if we find one).
Client would only need to import `chain/`.

Reply:
Do you want me to do that already ?

Reply:
seems reasonable to split that to a followup